conan_provider.cmake

```diff
@@ -14,18 +14,51 @@ function(detect_os OS OS_VERSION)
         else()
             set(${OS} ${CMAKE_SYSTEM_NAME} PARENT_SCOPE)
         endif()
+        if(${CMAKE_SYSTEM_NAME} STREQUAL "Android")
+            string(REPLACE "-" ";" PLATFORM_LIST ${ANDROID_PLATFORM})
```

Comment:
From the documentation, I can see that the `ANDROID_PLATFORM` can take 3 different formats. Looks like the logic here would work with `android-32` or just `32` - but maybe not for the one where only the letter is specified? I wonder if we need different cases based on regular expressions! 

Not a blocker by any beans if both `android-32` and `32` would both map to the correct value

Reply:
I pushed a fix that will accept `android-32` or `32`. Also, the toolchain changes the letter identifiers to the numbers so we don't have to deal with them. I added some test coverage and spot-checked and it looks better than it was, though it might be worth a closer look and more testing eventually.